Source/Moya.swift

```diff
@@ -21,7 +21,33 @@ public protocol TargetType {
     var path: String { get }
     var method: Moya.Method { get }
     var parameters: [String: AnyObject]? { get }
+    var parameterEncoding: ParameterEncoding { get }
     var sampleData: NSData { get }
+
+    func toEndpoint() -> Endpoint
+}
+
+public extension TargetType {
+    func toEndpoint() -> Endpoint {
+        let url = self.baseURL.URLByAppendingPathComponent(self.path).absoluteString
+        return Endpoint(URL: url, sampleResponseClosure: {.NetworkResponse(200, self.sampleData)}, method: self.method, parameters: self.parameters, parameterEncoding: parameterEncoding)
+    }
```

Comment:
Yes, that's my mistake, maybe we should create a function `convertToEndpoint` instead.
anyway, just don't let `Provider` do this.